### PR TITLE
Feature/aa 1942 add fields to booking summary report for agent

### PR DESCRIPTION
**Pull request by DaniyarAskarov**
Issue: https://github.com/happy-travel/agent-app-project/issues/1942

### CODE REVIEW

**Review by DFSko — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs [hunk 1]
```diff
@@ -161,6 +162,13 @@ static BookingSummaryNotificationData.BookingData CreateBookingData(Booking book
                         Status = EnumFormatters.FromDescription(booking.Status),
                         PaymentType = EnumFormatters.FromDescription(booking.PaymentType)
                     };
+
+
+                string GetAgentFormattedName(Booking booking)
+                    => _context.Agents
+                        .Where(a => a.Id == booking.AgentId)
+                        .Select(a => $"{a.FirstName} {a.LastName}")
```
Comment (AlexeyShok) [inline Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs:170]: What happens if an agent with this Id is not found?

**Review by AlexeyShok — APPROVED** [review]

File: Api/Models/Mailing/BookingSummaryNotificationData.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public class BookingSummaryNotificationData : DataWithCompanyInfo
         public class BookingData
```
Comment (suhan3z) [inline Api/Models/Mailing/BookingSummaryNotificationData.cs:14]: Lack of comments

File: Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs [hunk 3]
```diff
@@ -147,10 +147,17 @@ async Task<(List<EmailAndSetting>, List<Booking>)> GetBookings(List<EmailAndSett
                 return results;
 
 
-                static BookingSummaryNotificationData.BookingData CreateBookingData(Booking booking)
-                    => new()
+                BookingSummaryNotificationData.BookingData CreateBookingData(Booking booking)
+                {
+                    var agent = _context.Agents
```
Comment (suhan3z) [inline Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs:152]: Do we need to show only one agent?

**Review by suhan3z — CHANGES_REQUESTED** [review]
According to constants

File: Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs (hunk 3 above)
Comment (DaniyarAskarov) [reply Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs:152]: Per each booking has only one agent who booked it. According to task this agent's name should be added to mail template

File: Api/Models/Mailing/BookingSummaryNotificationData.cs (hunk 2 above)
Comment (DaniyarAskarov) [reply Api/Models/Mailing/BookingSummaryNotificationData.cs:14]: This class and mail templates were already created without comments. If we need to apply comments to already existed functional, we need to create separate task to cover the whole application with comments.

Comment (evm-andrey) [reply Api/Models/Mailing/BookingSummaryNotificationData.cs:14]: > This class and mail templates were already created without comments. If we need to apply comments to already existed functional, we need to create separate task to cover the whole application with comments.

You can start leaving comments little by little within the framework of the current task.

File: Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs (hunk 3 above)
Comment (suhan3z) [reply Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs:152]: If it's only one, then why we try to find him through firstOrDefault?

**Review by suhan3z — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs (hunk 3 above)
Comment (DaniyarAskarov) [reply Api/Services/Accommodations/Bookings/Mailing/BookingReportsService.cs:152]: I changed it SingleOrDefault. But it still one agent who booked it. I get it from db which contain the whole bunch of agents

**Pull request merged**